parser.go

```diff
@@ -2293,85 +2300,121 @@ func (p *Parser) lookaheadSimpleType() bool {
 //
 // ================================================================================
 
-func (p *Parser) parseDDL() ast.DDL {
+func (p *Parser) parseDDL() (ddl ast.DDL) {
+	l := p.Lexer.Clone()
+	defer func() {
+		if r := recover(); r != nil {
+			ddl = p.handleParseStatementError(r, l)
+		}
+	}()
+
 	pos := p.Token.Pos
 	switch {
 	case p.Token.Kind == "CREATE":
 		p.nextToken()
 		switch {
 		case p.Token.IsKeywordLike("SCHEMA"):
-			return p.parseCreateSchema(pos)
+			ddl = p.parseCreateSchema(pos)
+			return
```

Comment:
I think explicit return is preferred than naked returns even if you use named result parameters.

https://go.dev/wiki/CodeReviewComments#named-result-parameters
> Naked returns are okay if the function is a handful of lines. Once it’s a medium sized function, be explicit with your return values.